lib/rules/no-uppercase-with-underscore-property-name.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+const INVALID_PROPERTY_NAME_REGEX = /(?=.*_).*[A-Z]/;
+
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            description: 'Disallow property names containing both uppercase and underscores.',
+            category: 'LWC',
+            recommended: true,
+            url: docUrl('no-uppercase-with-underscore-property-name'),
+        },
+
+        schema: [],
+    },
+
+    create(context) {
+        return {
+            Identifier(node) {
```

Comment:
This does match against all the `identifier` in the code and not only the public properties. The linting rule will report an error in the following code which is incorrect:

```js
const foo_Bar = 1;
```

You can find detail on how to scope the rule to only run on public properties here: https://github.com/salesforce/eslint-plugin-lwc/blob/master/lib/rules/valid-api.js#L127